### PR TITLE
New run_type nat

### LINKED ISSUE #122: [Feature Request] Add TPROXY support
- [x] I certify that I acknowledge if I don't follow the format below or I don't check this box, my issue will be closed immediately without any notice.

It would be nice to have tproxy support to accept traffic from iptables, in this case it could act as a transparent proxy for LAN network. 

reference:
https://www.kernel.org/doc/Documentation/networking/tproxy.txt

### PR DESCRIPTION
**Pull request by GreaterFire**
Closes #122

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -44,31 +50,33 @@ if(OPENSSL_VERSION VERSION_GREATER_EQUAL 1.1.1)
...
     target_link_libraries(trojan ${MYSQL_LIBRARIES})
     add_definitions(-DENABLE_MYSQL)
 endif()
+
 option(FORCE_TCP_FASTOPEN "Force build with TCP Fast Open support" OFF)
 if(FORCE_TCP_FASTOPEN)
     add_definitions(-DTCP_FASTOPEN=23 -DTCP_FASTOPEN_CONNECT=30)
 endif()
+
+if(CMAKE_HOST_SYSTEM_NAME STREQUAL Linux)
```
Comment (wongsyrone) [inline CMakeLists.txt]: https://cmake.org/cmake/help/latest/variable/CMAKE_HOST_SYSTEM_NAME.html

On systems that have the uname command, this variable is set to the output of uname -s, thus not cross-compiling friendly.

Comment (wongsyrone) [reply CMakeLists.txt]: I prefer to `CMAKE_SYSTEM_NAME`

https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_NAME.html
https://github.com/openwrt/openwrt/blob/master/include/cmake.mk#L47

**Comment by wongsyrone** [comment]
BTW, can we add `-D_CRT_SECURE_NO_WARNINGS` for MSVC build although it's irrelevant to this PR.

```cmake
if(MSVC)
    add_definitions(-D_CRT_SECURE_NO_WARNINGS)
endif()
```

File: CMakeLists.txt (hunk 1 above)
Comment (GreaterFire) [reply CMakeLists.txt]: So should we use [`CMAKE_SYSTEM_NAME`](https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_NAME.html)?

**Comment by GreaterFire** [comment]
Sure. I'll add it directly to the `dev` branch or you can do it as well.

File: CMakeLists.txt (hunk 1 above)
Comment (wongsyrone) [reply CMakeLists.txt]: Got another method but it doesn't detect Linux directly.

```cmake
if(UNIX AND NOT APPLE)
    # for Linux, BSD, Solaris, Minix
endif()
```

Most of the existing solutions use CMAKE_SYSTEM_NAME

Comment (GreaterFire) [reply CMakeLists.txt]: I don't like the `UNIX AND NOT APPLE`.

Comment (GreaterFire) [reply CMakeLists.txt]: Anyway, I force pushed and resolved this.

**Comment by wongsyrone** [comment]
> Sure. I'll add it directly to the `dev` branch or you can do it as well.

Just push to `dev`.

**Pull request merged**